test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+        /// </param>
+        /// <param name="outDensifyResolution">The value used to densify the resulting geometry. Double.NaN for no densification.</param>
+        /// <param name="isAttFixOutInvPolygons">True if the process should attempt to fix invalid geometry, False otherwsie.</param>
+        /// <returns>
+        /// A Geometry class that contains the result of converting the coordinates into a Polyline or Polygon. If the geometry crosses the Dateline
+        /// then the result is multiple geometries wrt to the Eastern and Western hemisphere.
+        /// </returns>
+        /// <remarks>
+        /// Code by Abel G. Perez (Dec 19, 2021)
+        /// Todo: This algorithm assumes the polygon is small to medium in width. Typically if the width of the polygon is <180° then the algorithm
+        /// should work well. This is because the core of the process shifts all coordinates to either the Eastern or Western hemisphere. A better
+        /// approach may be to shift all coordinates to the 0° to +360° range. This will allow polygons with larger widths.  This will get implemented
+        /// in a future update if required. However, it really depends on how the user inputs the coordinates. If the input already has coordinates in
+        /// the 0° to +360° range then that is straightforward to detect when a geometry crosses the dateline. But if the inputs are all in the standard
+        /// -180° to 0° to +180° range then the assumption has to be made if the geometry closes towards the prime meridian or towards the dateline.
+        /// The current algorithm determines this by simply calculting which critical meridian is the closest.
+        /// https://github.com/NetTopologySuite/NetTopologySuite/discussions/496
+        /// https://github.com/NetTopologySuite/NetTopologySuite/blob/develop/test/NetTopologySuite.Samples.Console/Operation/Polygonize/SplitPolygonExample.cs
+        /// https://github.com/NetTopologySuite/NetTopologySuite/blob/84aef3c6a6cd3c2bfa5e3034ba161555cef5a117/test/NetTopologySuite.Samples.Console/Operation/Polygonize/SplitPolygonExample.cs#L24
+        /// https://github.com/nettopologysuite/nettopologysuite/issues/129
+        /// https://github.com/DotSpatial/DotSpatial/blob/e30421d6ee5b0cdf2b1fa01f95f8aabc50c1997a/Source/DotSpatial.Data/Shape.cs#L611
+        /// https://github.com/DotSpatial/DotSpatial/issues/1029
+        /// https://github.com/NetTopologySuite/ProjNet4GeoAPI/wiki/Well-Known-Text
+        /// https://desktop.arcgis.com/en/arcmap/10.3/manage-data/editing-fundamentals/creating-and-editing-multipart-polygons.htm
+        /// https://desktop.arcgis.com/en/arcmap/10.3/guide-books/map-projections/what-happens-to-features-at-180-dateline-.htm
+        /// </remarks>
+        public static Geometry ToPolyExOp(List<Coordinate> inpCoords, string inpProj4Wkt, OgcGeometryType outType, double outTrimGap, double outDensifyResolution, bool isAttFixOutInvPolygons)
```

Comment:
"out" parameters (all the parameters following "inpProj4Wkt") can be moved to a single "parameter class", just for the sake of the readability

Reply:
Question, is there a number of parameters where you move to a parameter class? Obviously if you have 30 params you don't want those all in the function call. If I have 3 or 4, I typically leave it in the function call as-is. Looking at the NTS core code there are various calls with 5 even 6 params right in the function call. So I'm wondering if this change is needed? I'm open to it.

Reply:
Simple a suggestion, just because there are several "out" parameters, that _maybe_ can be grouped for the sake of the readability.
Totally unrelated, see something like JsonConvert.Deserialize(jsonString, jsonDeserializationParamsClass).

Reply:
Ok I will take a loo. Thanks.